File: python/dsflix_sdk/modules/membership.py
from __future__ import annotations

from ..http import HttpClient


class MembershipModule:
    def __init__(self, http: HttpClient) -> None:
        self.http = http

    def get_plans(self) -> dict:
        return self.http.get("/api/v2/membership/plans")

    def get_status(self) -> dict:
        return self.http.get("/api/v2/membership/status")

    def upgrade(self, plan: str, duration_days: int, payment_method: str | None = None) -> dict:
        return self.http.post(
            "/api/v2/membership/upgrade",
            {"plan": plan, "duration_days": duration_days, "payment_method": payment_method},
        )

File: src/modules/membership.ts
// ─────────────────────────────────────────────────────────────
//  DSFlix SDK  –  src/modules/membership.ts
// ─────────────────────────────────────────────────────────────

import { HttpClient } from '../http';
import {
  MembershipPlansResponse,
  MembershipStatus,
  MembershipUpgradeParams,
  MembershipUpgradeResponse,
} from '../types';

export class MembershipModule {
  constructor(private http: HttpClient) {}

  /** Get all available membership plans */
  getPlans(): Promise<MembershipPlansResponse> {
    return this.http.get('/api/v2/membership/plans');
  }

  /** Get the current user's membership status */
  getStatus(): Promise<MembershipStatus> {
    return this.http.get('/api/v2/membership/status');
  }

  /**
   * Upgrade the authenticated user to Pro or Plus plan.
   * @param plan - 'pro' or 'plus'
   * @param durationDays - Duration (min 7, max 365)
   * @param paymentMethod - 'coins' or a Stripe payment_method_id
   */
  upgrade(plan: 'plus' | 'pro', durationDays: number, paymentMethod?: string): Promise<MembershipUpgradeResponse> {
    const params: MembershipUpgradeParams = { plan, duration_days: durationDays, payment_method: paymentMethod };
    return this.http.post('/api/v2/membership/upgrade', params);
  }
}
